p8e-util/src/main/kotlin/io/p8e/crypto/SignatureInputStream.kt

```diff
@@ -1,21 +1,64 @@
 package io.p8e.crypto
 
 import java.io.InputStream
-import java.security.Signature
+import java.security.PublicKey
 
 class SignatureInputStream(
     private val inputStream: InputStream,
-    private val signature: Signature
+    private val signer: SignerImpl,
+    private val signatureBytes: ByteArray? = null
 ): InputStream() {
     override fun read(): Int {
         val res = inputStream.read()
         if (res != -1) {
-            signature.update(res.toByte())
+            signer.update(inputStream.readAllBytes())
```

Comment:
I assume this starts from the beginning of the stream (i.e. the read call above didn't advance one byte into the stream or something like that?)

Reply:
don't think this read method gets invoked, the other read with the off and len does, didn't spend anytime time looking if the other 2 reads are used for something else

Reply:
crap .. yea, i have to revert that too